contracts/src/v0.8/dev/ocr2dr/OCR2Base.sol

```diff
@@ -0,0 +1,372 @@
+// SPDX-License-Identifier: MIT
```

Comment:
Why this file is copied here? Is it a different implementation?
`OCR2Base.sol` is located `../ocr2/OCR2Base.sol`

Reply:
See the note in the description:

> contracts/src/v0.8/dev/ocr2/OCR2Base.sol has been slightly modified and copied into the ocr2dr folder.
Please see [here](https://github.com/smartcontractkit/chainlink/compare/jk/ocr2dr-billing-2-base-diff?expand=1) for a nicer to read diff.

Reply:
I would not hesitate to modify the original OCR2Base.sol. Your changes look reasonable and backward compatible. Also, AFAIK nobody else uses this base contract at the moment.

Reply:
Yeah, I believe it is just an example contract.

It is actually weird to me that in the current OCR2Base.sol `_report` is being called before the signer validation.
Even without the billing specific changes I believe we would want to modify this. Otherwise the OCR2DR callback can be executed with data that has not had DON consensus.

Reply:
I find that call order weird too, we should ask Research if there is any reasoning behind it.